cpp/src/parquet/encoding.cc

```diff
@@ -904,7 +904,12 @@ class DecoderImpl : virtual public Decoder {
 
  protected:
   explicit DecoderImpl(const ColumnDescriptor* descr, Encoding::type encoding)
-      : descr_(descr), encoding_(encoding), num_values_(0), data_(NULLPTR), len_(0) {}
+      : descr_(descr),
+        encoding_(encoding),
+        num_values_(0),
+        data_(NULLPTR),
+        len_(0),
+        type_length_(0) {}
```

Comment:
Could you get `type_lenght` from the `ColumnDescriptor` for fixed width types? I don't remember if type is available here or not.

Reply:
Revert this :-(

This only works in FLBA, just ignore it...